### PR TITLE
Feat: Added dark mode functionality

### PR DESCRIPTION
**Pull request by Mayor-Rhymes**
@ArinzeJeffrey-droid. I added dark mode functionality to the website and it was suspiciously too easy. Please check your whatsapp for the demo. Thank you so much.

### CODE REVIEW

File: src/components/themeprovider.tsx [hunk 1]
```diff
@@ -0,0 +1,8 @@
+"use client";
```
Comment (ArinzeJeffrey-droid) [inline src/components/themeprovider.tsx:1]: I genuinely don't know what this `"use client"` do

Comment (Mayor-Rhymes) [reply src/components/themeprovider.tsx:1]: > I genuinely don't know what this `"use client"` do
It is a very weird keyword that next js 13 app router introduced. If I don't add it, I will not be able to use client hooks like useState, useEffect and the likes. It signifies to the compiler that this component will be dynamic.

Comment (Mayor-Rhymes) [reply src/components/themeprovider.tsx:1]: > I genuinely don't know what this `"use client"` do

It is a very weird keyword that next js 13 app router introduced. If I don't add it, I will not be able to use client hooks like useState, useEffect and the likes. It signifies to the compiler that this component will be dynamic.

Comment (ArinzeJeffrey-droid) [reply src/components/themeprovider.tsx:1]: AHHHHH makes sense.
Thanks for clarifying 🙏🏾

Comment (Mayor-Rhymes) [reply src/components/themeprovider.tsx:1]: @ArinzeJeffrey-droid Additionally, there is also one called 'use server', it is a keyword that tells the compiler that the component is a react server component and therefore, will be able to directly fetch data. TLDR, a react component can become asynchronous and will have the ability to directly access database and do things that are asynchronous.

**Review by ArinzeJeffrey-droid — APPROVED** [review]

**Pull request merged**